core/src/test/java/org/apache/iceberg/catalog/CatalogTests.java

```diff
@@ -2615,6 +2616,100 @@ public void tableCreationWithoutNamespace() {
         .hasMessageContaining("Namespace does not exist: non-existing");
   }
 
+  @Test
+  public void testRegisterTable() {
+    C catalog = catalog();
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(TABLE.namespace());
+    }
+
+    Map<String, String> properties =
+        ImmutableMap.of("user", "someone", "created-at", "2023-01-15T00:00:01");
+    Table table =
+        catalog
+            .buildTable(TABLE, SCHEMA)
+            .withPartitionSpec(SPEC)
+            .withSortOrder(WRITE_ORDER)
+            .withProperties(properties)
+            .create();
+
+    table.newFastAppend().appendFile(FILE_A).commit();
+    table.newFastAppend().appendFile(FILE_B).commit();
+    table.newDelete().deleteFile(FILE_A).commit();
+    table.newFastAppend().appendFile(FILE_C).commit();
+
+    TableOperations ops = ((HasTableOperations) table).operations();
+    String metadataLocation = ops.current().metadataFileLocation();
+
+    catalog.dropTable(TABLE, false);
+
+    Table registeredTable = catalog.registerTable(TABLE, metadataLocation);
+    Assertions.assertThat(registeredTable).isNotNull();
+
+    TableOperations registeredTableOps = ((HasTableOperations) table).operations();
+    Assertions.assertThat(metadataLocation)
+        .isEqualTo(registeredTableOps.current().metadataFileLocation());
+
+    Map<String, String> actualProperties = Maps.newHashMap(table.properties());
+    Map<String, String> expectedProperties = Maps.newHashMap(table.properties());
+    // The nessie.commit.id for each table will be distinct, hence, it is necessary to remove it
+    // before asserting the properties.
```

Comment:
`CatalogTests` should not have anything specific to Nessie. Instead of doing this, I think the validation should check that the properties used to create the table are a subset of the properties in the actual table.